### PR TITLE
Harden our Check for Common-Leading-Whitespace in Doc-Comments

### LINKED ISSUE #800: Doc-Comment Indent Stripping Splits a UTF-8 Codepoint
`comments/grammar.rs:105–125`: the common leading-whitespace amount is computed as a byte offset via `str::find`, min'd across lines, then applied to every line with `replace_range(..common, "")`.

If lines have different-byte-width leading whitespace (an ASCII space on one line, a multi-byte Unicode space like NBSP U+00A0 on another — a common copy-paste artifact), the byte offset lands mid-codepoint → panic `end of range should be a character boundary`. Parsed at construction time, before validation:

## Reproduction

```
printf '/// a\n///\xc2\xa0b\nstruct S {}\n' > nbsp.slice && ./slicec nbsp.slice --dry-run
```

It's also a latent correctness bug (mixed indentation stripped by the wrong amount).

## Fix

Measure and strip in chars, not bytes: `char_indices().nth(common)`.

### PR DESCRIPTION
**Pull request by InsertCreativityHere**
This PR is a code-hardening to handle doc-comments where a user has different kinds of whitespace at the starts of their doc-comment lines. It fixes #800.

----

It is common for users to have whitespace between a `///` and the actual comment's text:
```
/// See how there's a space after '///' here?.
///     - This line could be indented as a bullet point.
///And this line has no leading whitespace at all.
```
We have code to detect this and strip it away from the comment's actual content.
The exact check is to determine the "common leading whitespace" and remove it; i.e. find the maximum amount of whitespace such that every line starts with at least that amount of whitespace.

However, this check isn't very exact right now. We currently only count the _number_ of whitespace characters. So if I have a multiline comment, one starts with 2 spaces, and one starts with 2 tabs, we'd count those as the same amount of whitespace, which is wrong. If you have 2 spaces and 2 non-ASCII whitespace characters, this could ever cause a panic, since we're counting over _bytes_ instead of unicode characters.

This PR changes this so we track the `whitespace` as an actual `str`, not just a `count` of characters. We determine the largest substring such that:
- It is only made of whitespace characters
- Every single line of a comment starts with it.

And like before, we strip it from every single line.

## What's Changed

Nothing - this only affects doc-comments where an inconsistent kind of whitespace is used to pad the start of each line.
In practice, almost all users will use a consistent kind of whitespace padding.

### CODE REVIEW

File: slicec/src/parsers/comments/grammar.rs [hunk 1]
```diff
@@ -90,41 +90,49 @@ fn construct_section_message(
...
+        // then this line has no leading whitespace, so the _common_ leading whitespace must be empty.
+        let MessageComponent::Text(text) = first_component else {
+            common_leading_whitespace = Some("");
+            break;
+        };
+
+        common_leading_whitespace = Some(match common_leading_whitespace {
+            Some(current_whitespace) => get_common_prefix(current_whitespace, text),
+            None => get_leading_whitespace(text),
+        });
```
[the remark this answers is not in the document]
Comment (InsertCreativityHere) [reply slicec/src/parsers/comments/grammar.rs:109]: I think this is minor and finicky enough to not warrant a test, if any other reviewers disagree, let me know and I will add a test using a mix of ASCII and non-ASCII whitespace at the start of doc-comment lines.

**Review by externl — APPROVED** [review]
Looks good!

**Review by pepone — APPROVED** [review]
